features/attach_invalidtoken.feature

```diff
@@ -18,7 +18,7 @@ Feature: Command behaviour when trying to attach a machine to an Ubuntu
         When I verify that running `ua attach invalid-token --format json` `with sudo` exits `1`
         Then I will see the following on stdout:
             """
-            {"_schema_version": "0.1", "errors": [{"message": "Invalid token. See https://ubuntu.com/advantage", "service": null, "type": "system"}], "failed_services": [], "needs_reboot": false, "processed_services": [], "result": "failure", "warnings": []}
+            {"_schema_version": "0.1", "errors": [{"message": "Invalid token. See https://ubuntu.com/advantage", "message_code": "attach-invalid-token", "service": null, "type": "system"}], "failed_services": [], "needs_reboot": false, "processed_services": [], "result": "failure", "warnings": []}
```

Comment:
Instead of making the `message_code` numbers, I have decided to turn them into strings, since I think it will be to rely on strings in the Desktop/Subiquity code than on numbers.

If agree with that, one change we can do here is to create constants for the `message_codes` so developers can import then directly from ua

Reply:
I completely agree with message codes as strings.

The idea of exposing the string constants is a good one, and I think we should talk about this more as we discuss making something like a "public" api for uaclient, but probably not include it in this PR.